src/api/code-challenge/content-types/code-challenge/lifecycles.js

```diff
@@ -0,0 +1,92 @@
+module.exports = {
+  beforeCreate(event) {
+    console.log("**** beforeCreate hook is activated! ****");
+    const eventData = event.params.data;
+    runMetaTest(eventData);
+  },
+
+  beforeCreateMany(event) {
+    console.log("**** beforeCreateMany hook is activated! ****");
+    const eventData = event.params.data;
+    runMetaTest(eventData);
+  },
+
+  beforeUpdate(event) {
+    console.log("**** beforeUpdate hook is activated! ****");
+    const eventData = event.params.data;
+    runMetaTest(eventData);
+  },
+
+  beforeUpdateMany(event) {
+    console.log("**** beforeUpdateMany hook is activated! ****");
+    const eventData = event.params.data;
+    runMetaTest(eventData);
+  },
+};
+
+async function runMetaTest(eventData) {
+  const tests = await getInternalTests(eventData);
+  const metaTests = await getMetaTests(eventData);
+
+  tests.forEach((test) =>
+    console.log("final test.internalTest: ", test.internalTest)
+  );
+
+  metaTests.forEach((metaTest) =>
+    console.log("final metaTest caseCode: ", metaTest.caseCode)
+  );
+
+  return false;
+}
+
+async function getInternalTests(eventData) {
+  // Original Finding Internal Test
+  const internalTests = await strapi.db
+    .query("challenge.code-challenge-test")
+    .findMany({
+      select: ["id", "internalTest"],
+      where: {
+        id: {
+          $gte: getCurrentTestID(eventData),
```

Comment:
what does this stand for? tool lazy to look it up

Reply:
It is "greater than or equal to" for filtering db.query in strapi. used with "where" keyword

Reply:
$gte has been replaced with an array, to select the exact IDs needed